Presentation/UIViewController+Presentation.swift

```diff
@@ -63,9 +69,11 @@ public extension UIViewController {
             }
 
             bag += {
-                guard !didComplete else { return }
-                log(.didCancel(.init(vc.presentationDescription), from: .init(self.presentationDescription)))
-                completion(.failure(PresentError.dismissed))
+                originalScheduler.async {
```

Comment:
Ok @moglistree, I ran the test using the thread sanitiser, and the problem is both in the log (title access) and the dismiss as you mentioned. For this to work we need to be on main so I think making that explicit is better, as well as adding a comment:

```swift
Scheduler.main.async { // Disposing gives no guarantees on which thread it is run.
```

But I also like the idea of adding the helper to Flow mentioned in https://github.com/iZettle/Flow/pull/89:

```swift
extension Disposer
 convenience init(on scheduler: Scheduler, _ disposer: @escaping () -> Void) {
   self.init { scheduler.async(disposer) }
 }
}
```

so the above can be written:

```swift
bag += Disposer(on: .main) {

}
```

Reply:
The flow helper we will take in another PR but I agree the main thread should be defaulted in the disposer. Will make the update now

Reply:
Or would u prefere solving this by updating flow first?

Reply:
Not necessary.